### PR TITLE
Disable token chain and token inputs when using a reward template

### PR DESCRIPTION
**Pull request by Devorein**
### WHY

[Card](https://app.charmverse.io/charmverse/disable-token-chain-and-token-inputs-when-using-a-reward-template-8641440715535329)

### CODE REVIEW

File: components/proposals/ProposalPage/components/ProposalProperties/components/ProposalRewards/ProposalRewardsForm.tsx [hunk 1]
```diff
@@ -57,16 +52,18 @@ export function ProposalRewardsForm({
...
+  rewardStatus
 }: Props) {
   const { getFeatureTitle } = useSpaceFeatures();
-  const isAdmin = useIsAdmin();
   const [isExpanded, setIsExpanded] = useState(!!expandedByDefault);
   const { data: template } = useGetRewardTemplate(templateId);
-  const readOnlyProperties = !isAdmin && (readOnly || !!template);
-  const readOnlyDueDate = !isAdmin && (readOnly || !!template?.dueDate);
-  const readOnlyToken = !isAdmin && (readOnly || !!template?.rewardToken);
+  const readOnlyType = readOnly || !!template?.rewardType;
```
Comment (mattcasey) [inline components/proposals/ProposalPage/components/ProposalProperties/components/ProposalRewards/ProposalRewardsForm.tsx:60]: admins are always supposed to be able to override template settings

File: components/proposals/ProposalPage/components/ProposalProperties/components/ProposalRewards/ProposalRewardsForm.tsx [hunk 2]
```diff
@@ -57,16 +52,18 @@ export function ProposalRewardsForm({
...
+  const readOnlyDueDate = readOnly || !!template?.dueDate;
+  const readOnlyToken = readOnly || !!template?.rewardToken;
+  const readOnlyChain = readOnly || !!template?.chainId;
```
Comment (mattcasey) [inline components/proposals/ProposalPage/components/ProposalProperties/components/ProposalRewards/ProposalRewardsForm.tsx:63]: i dont think we actually have a use case where only chain is set but token is optional ,so i dont think we should support it. if the token's read only, then chain is readonly

File: components/proposals/ProposalPage/components/ProposalProperties/hooks/useProposalRewards.ts [hunk 3]
```diff
@@ -68,7 +68,7 @@ export function useProposalRewards({
   }).join(', ');
 
   function getTemplateAppliedReward(template?: RewardTemplate) {
-    const rewardType = template ? getRewardType(template) : 'token';
+    const rewardType = template ? template.rewardType : 'token';
```
Comment (mattcasey) [inline components/proposals/ProposalPage/components/ProposalProperties/hooks/useProposalRewards.ts:71]: thx for cleanup

**Review by mattcasey — CHANGES_REQUESTED** [review]
left some comments. blocking issue is that admins cannot override templates

File: components/proposals/ProposalPage/components/ProposalProperties/components/ProposalRewards/ProposalRewardsForm.tsx (hunk 1 above)
Comment (Devorein) [reply components/proposals/ProposalPage/components/ProposalProperties/components/ProposalRewards/ProposalRewardsForm.tsx:60]: My bad I thought this restriction will be for every user (including admin). Added admin override.

File: components/proposals/ProposalPage/components/ProposalProperties/components/ProposalRewards/ProposalRewardsForm.tsx (hunk 2 above)
Comment (Devorein) [reply components/proposals/ProposalPage/components/ProposalProperties/components/ProposalRewards/ProposalRewardsForm.tsx:63]: Sure lets go with that.

**Pull request merged**